openssl-sys/src/handwritten/pkcs7.rs

```diff
@@ -1,12 +1,195 @@
+        #[repr(C)]
+        pub struct PKCS7_SIGNED {
+            pub version: *mut ASN1_INTEGER, /* version 1 */
+            pub md_algs: *mut stack_st_X509_ALGOR, /* md used */
+            pub cert: *mut stack_st_X509, /* [ 0 ] */
+            pub crl: *mut stack_st_X509_CRL, /* [ 1 ] */
+            pub signer_info: *mut stack_st_PKCS7_SIGNER_INFO,
+            pub contents: *mut PKCS7,
+        }
+    } else {
+        pub enum PKCS7_SIGNED {}
+    }
+}
+
+cfg_if! {
+    if #[cfg(any(ossl101, libressl251))] {
+        #[repr(C)]
+        pub struct PKCS7_ENC_CONTENT {
+            pub content_type: *mut ASN1_OBJECT,
+            pub algorithm: *mut X509_ALGOR,
+            pub enc_data: *mut ASN1_OCTET_STRING, /* [ 0 ] */
+            pub cipher: *const EVP_CIPHER,
+            #[cfg(ossl300)]
+            pub ctx: *const PKCS7_CTX,
+       }
+    } else {
+        pub enum PKCS7_ENC_CONTENT {}
+    }
+}
+
+cfg_if! {
+    if #[cfg(any(ossl101, libressl251))] {
+        #[repr(C)]
+        pub struct PKCS7_ENVELOPE {
+            pub version: *mut ASN1_INTEGER, /* version 0 */
+            pub recipientinfo: *mut stack_st_PKCS7_RECIP_INFO,
+            pub enc_data: *mut PKCS7_ENC_CONTENT,
+        }
+    }  else {
+        pub enum PKCS7_ENVELOPE {}
+    }
+}
+
+cfg_if! {
+    if #[cfg(any(ossl101, libressl251))] {
+        #[repr(C)]
+        pub struct PKCS7_SIGN_ENVELOPE {
+            pub version: *mut ASN1_INTEGER, /* version 1 */
+            pub md_algs: *mut stack_st_X509_ALGOR, /* md used */
+            pub cert: *mut stack_st_X509, /* [ 0 ] */
+            pub crl: *mut stack_st_X509_CRL, /* [ 1 ] */
+            pub signer_info: *mut stack_st_PKCS7_SIGNER_INFO,
+            pub enc_data: *mut PKCS7_ENC_CONTENT,
+            pub recipientinfo: *mut stack_st_PKCS7_RECIP_INFO
+        }
+    } else {
+        pub enum PKCS7_SIGN_ENVELOPE {}
+    }
+}
+
+cfg_if! {
+    if #[cfg(any(ossl101, libressl251))] {
+        #[repr(C)]
+        pub struct PKCS7_DIGEST {
+            pub version: *mut ASN1_INTEGER, /* version 0 */
+            pub md: *mut X509_ALGOR, /* md used */
+            pub contents: *mut PKCS7,
+            pub digest: *mut ASN1_OCTET_STRING,
+        }
+    } else {
+        pub enum PKCS7_DIGEST {}
+    }
+}
+
+cfg_if! {
+    if #[cfg(any(ossl101, libressl251))] {
+        #[repr(C)]
+        pub struct PKCS7_ENCRYPT {
+            pub version: *mut ASN1_INTEGER, /* version 0 */
+            pub enc_data: *mut PKCS7_ENC_CONTENT,
+        }
+    } else {
+        pub enum PKCS7_ENCRYPT {}
+    }
+}
+
+extern "C" {
+    pub fn PKCS7_SIGNED_free(info: *mut PKCS7_SIGNED);
+    pub fn PKCS7_ENC_CONTENT_free(info: *mut PKCS7_ENC_CONTENT);
+    pub fn PKCS7_ENVELOPE_free(info: *mut PKCS7_ENVELOPE);
+    pub fn PKCS7_SIGN_ENVELOPE_free(info: *mut PKCS7_SIGN_ENVELOPE);
+    pub fn PKCS7_DIGEST_free(info: *mut PKCS7_DIGEST);
+    pub fn PKCS7_SIGNER_INFO_free(info: *mut PKCS7_SIGNER_INFO);
```

Comment:
`free` functions appear to be missing for `PKCS7_ENCRYPT` and `PKCS7_ISSUER_AND_SERIAL`.

Reply:
Added.